tests/run/carray_slicing.pyx

```diff
@@ -275,3 +276,227 @@ def struct_ptr_iter():
     return ([ value.i for value in my_structs[:5] ],
             [ ptr.i for ptr in my_structs[:5] ],
             [ inferred.i for inferred in my_structs[:5] ])
+
```

Comment:
I can't see any tests for wraparound although your code claims to handle it.

Reply:
It looks like the tests you've added are for boundschecking (which is good - this should be tested). You should probably test `a[:-5]` or `a[-2:]` or `a[-5:-2]` or `a[-5:8]` etc.